### PR TITLE
feat[v7]: support interchain security in local-interchain

### PR DESCRIPTION
**Pull request by Reecepbcups**
request by timewave

## Summary
Adds ICS support to local-interchain in a user friendly way with only 1 configuration change needed. Supports multiple consumers -> 1 provider

## Integration
Add `"ics_consumer_link": "providerchainid-1",` to your consumer chain. No changes on the provider side is required

## Run
`make install && local-ic start interchainsecurity`

### CODE REVIEW

**Comment by Reecepbcups** [comment]
Currently  `make install && local-ic start interchainsecurity` starts but then fails on the gov prop `failed to start provider chain localcosmos-1: proposal status did not change to passed in expected number of blocks: proposal status (PROPOSAL_STATUS_VOTING_PERIOD) does not match expected: (PROPOSAL_STATUS_PASSED)"`

I assume the genesis params are wrong given SDK v45. That should resolve it

**Comment by Reecepbcups** [comment]
StartConsumer seems to override any consumer chain genesis configs passed. Should modify the gensis with our params after the ccv setup consumer side (interchaintest bug)

File: local-interchain/interchain/start.go [hunk 1]
```diff
@@ -137,6 +144,40 @@ func StartChain(installDir, chainCfgFile string, ac *AppConfig) {
...
+			logger.Info("Adding ICS pair", zap.String("provider", p.Config().ChainID), zap.String("consumer", c.Config().ChainID), zap.String("path", pathName))
+
+			ic = ic.AddProviderConsumerLink(interchaintest.ProviderConsumerLink{
+				Provider: p,
+				Consumer: c,
+				Relayer:  relayer,
+				Path:     pathName,
+			})
+		}
+	}
```
Comment (jtieri) [inline local-interchain/interchain/start.go:179]: this could be its own function, which would increase readability and make it easier to maintain.

**Review by boojamya — APPROVED** [review]

**Review by jtieri — APPROVED** [review]
looks good to me! i like the one config change made to signal a chain is a consumer chain, may steal this for the relayer as i think i made two changes for ICS related functionality.

File: local-interchain/interchain/start.go (hunk 1 above)
Comment (Reecepbcups) [reply local-interchain/interchain/start.go:179]: atm many places in the `StartChain` could use this. It's kinda a monolith for starting up. Need to do a refactor here in the near future

**Pull request merged**